walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,114 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
 
 public final class FindMeetingQuery {
+
+  // create a set of events to avoid duplicates
+  private Collection<Event> events = new HashSet<>();
+
+  // the meeting request
+  private MeetingRequest request; 
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+ 
+    this.events = events;
+    this.request = request; 
+
+    // Check if duration is longer than a day
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList(); 
+    }
+    // Check for no events or no attendees
+    if (events.isEmpty()) {
+        if(request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
+            return Arrays.asList((TimeRange.WHOLE_DAY)); 
+        }
+    }
+
+    // Arraylist holds event times accordingly     
```

Comment:
"accordingly" to what? Let's update this to be more explicit about what that means.